pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -51,13 +58,127 @@ func (r *RedisOptionsProvider) GetRedisOptions() (*component.RedisOptions, error
 
 	r.setPersistentVolumeClaimOptions()
 
-	err := r.options.Validate()
+	// Should the operator be reading redis secrets?
```

Comment:
I think so.

Even if the DBs are deployed by the operator we allow the users to precreate the secret to allow non-default values (For example, using a specific password that the user wants to define instead of an auto-generated one)

Reply:
But then we are allowing the users to point to external redis instance (both backend and system) or even broken (non existing) redis instance, hence breaking the deployment. 

If the operator deploys redis internally, it should make sure backend and system are configured to use that redis. For that, "desired" url for the secret should be internal url. In case the customer accidentally updates that secret, the operator would reconcile to the internal redis, making the system more robust.

Are we going to allow custom password, allowing also broken deployment, when redis is deployed internally?

Reply:
When redis is deployed internally using a custom pre-creatd password should not break the deployent.

Reply:
As I commented in https://github.com/3scale/3scale-operator/pull/486#discussion_r500177300, until now we are assuming the user is responisble enough to not configure parameters that it shouldn't (like pointing to another DB when redis are internal), as we are doing with other configurations like resource requirements (we assume they place values that are enough), affinity/tolerations (we assume we configure them in compatible nodes architectures), and so on.

Reply:
leaving the comment as a reminder note for a future enhancement